File: packages/contracts/src/types/storage.ts
import type { AnalysisStatus, FindingSeverity } from "./analysis";
import type { ExportFormat } from "./report";

export type AuditLogLevel = "trace" | "debug" | "info" | "warn" | "error";

export interface StoredAnalysisRun {
  readonly runId: string;
  readonly requestId: string;
  readonly status: AnalysisStatus;
  readonly workflow: string;
  readonly imageId: string;
  readonly createdAt: string;
  readonly completedAt?: string;
  readonly errorCode?: string;
  readonly errorMessage?: string;
}

export interface StoredImageRecord {
  readonly imageId: string;
  readonly fileName: string;
  readonly sourcePath: string;
  readonly sha256: string;
  readonly mimeType: string;
  readonly width: number;
  readonly height: number;
  readonly sizeBytes: number;
  readonly importedAt: string;
}

export interface StoredModelRegistryEntry {
  readonly modelId: string;
  readonly name: string;
  readonly version: string;
  readonly task: "classification" | "detection" | "segmentation";
  readonly framework: "torchvision";
  readonly configPath: string;
  readonly checkpointPath?: string;
  readonly configHash: string;
  readonly checkpointHash?: string;
  readonly enabled: boolean;
}

export interface StoredReportArtifact {
  readonly artifactId: string;
  readonly runId: string;
  readonly format: ExportFormat;
  readonly path: string;
  readonly sha256: string;
  readonly sizeBytes: number;
  readonly createdAt: string;
}

export interface AuditLogEntry {
  readonly entryId: string;
  readonly level: AuditLogLevel;
  readonly eventType: string;
  readonly message: string;
  readonly runId?: string;
  readonly context?: Record<string, unknown>;
  readonly createdAt: string;
}

export interface FindingIndexRecord {
  readonly findingId: string;
  readonly runId: string;
  readonly severity: FindingSeverity;
  readonly confidence: number;
  readonly title: string;
}

File: engine/veriframe_core/veriframe_core/contracts/storage.py
from __future__ import annotations

from typing import Literal

from pydantic import Field

from veriframe_core.contracts.analysis import AnalysisStatus, FindingSeverity, StrictModel
from veriframe_core.contracts.report import ExportFormat

AuditLogLevel = Literal["trace", "debug", "info", "warn", "error"]


class StoredAnalysisRun(StrictModel):
    runId: str = Field(min_length=1)
    requestId: str = Field(min_length=1)
    status: AnalysisStatus
    workflow: str
    imageId: str
    createdAt: str
    completedAt: str | None = None
    errorCode: str | None = None
    errorMessage: str | None = None


class StoredImageRecord(StrictModel):
    imageId: str = Field(min_length=1)
    fileName: str
    sourcePath: str
    sha256: str = Field(pattern=r"^[a-fA-F0-9]{64}$")
    mimeType: str
    width: int = Field(ge=1)
    height: int = Field(ge=1)
    sizeBytes: int = Field(ge=0)
    importedAt: str


class StoredModelRegistryEntry(StrictModel):
    modelId: str
    name: str
    version: str
    task: Literal["classification", "detection", "segmentation"]
    framework: Literal["torchvision"]
    configPath: str
    checkpointPath: str | None = None
    configHash: str = Field(pattern=r"^[a-fA-F0-9]{64}$")
    checkpointHash: str | None = Field(default=None, pattern=r"^[a-fA-F0-9]{64}$")
    enabled: bool


class StoredReportArtifact(StrictModel):
    artifactId: str
    runId: str
    format: ExportFormat
    path: str
    sha256: str = Field(pattern=r"^[a-fA-F0-9]{64}$")
    sizeBytes: int = Field(ge=0)
    createdAt: str


class AuditLogEntry(StrictModel):
    entryId: str
    level: AuditLogLevel
    eventType: str
    message: str
    runId: str | None = None
    context: dict[str, object] | None = None
    createdAt: str


class FindingIndexRecord(StrictModel):
    findingId: str
    runId: str
    severity: FindingSeverity
    confidence: float = Field(ge=0, le=1)
    title: str
